persistence/provider/sql/mysql/offsetstore.go

```diff
@@ -0,0 +1,81 @@
+package mysql
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/dogmatiq/infix/eventstream"
+	"github.com/dogmatiq/infix/internal/x/sqlx"
+)
+
+// LoadOffset loads the last offset associated with the given source
+// application key sk. ak is the 'owner' application key.
+//
+// If there is no offset associated with the given source application key,
+// the offset is returned as zero and error as nil.
+func (driver) LoadOffset(
+	ctx context.Context,
+	db *sql.DB,
+	ak, sk string,
+) (eventstream.Offset, error) {
+	var o eventstream.Offset
+
+	row := db.QueryRowContext(
+		ctx,
+		`SELECT next_offset
+		FROM offset_store
+		WHERE app_key = ? AND source_app_key = ?`,
+		ak, sk,
+	)
+
+	if err := row.Scan(&o); err != nil {
+		if err == sql.ErrNoRows {
+			return 0, nil
+		}
+
+		return 0, err
+	}
+
+	return o, nil
+}
+
+// InsertOffset inserts a new offset associated with the given source
+// application key sk. ak is the 'owner' application key.
+//
+// It returns false if the row already exists.
+func (driver) InsertOffset(
+	ctx context.Context,
+	tx *sql.Tx,
+	ak, sk string,
+	n eventstream.Offset,
+) (bool, error) {
+	return insertIgnore(
+		ctx,
+		tx,
+		`INSERT INTO offset_store VALUES(?, ?, ?)`,
+		ak, sk, n,
```

Comment:
Please **always** use explicit column names in queries. Even though this is covered by tests within this repo, some way down the line someone will modify some production database as part of some weird hacky migration and individual queries will fail in weird and wonderful ways, especially with MySQL which seems to like to guess what you mean half the time :)

MySQL also has this "update-like" insert syntax, which I think is actually far more pleasant to work with than the `VALUES` tuple.

```suggestion
		`INSERT INTO offset_store SET
			app_key_id = ?,
			source_app_key = ?,
			next_offset = ?`,
		ak,
		sk,
		n,
```